crossdock/client/ctxpropagation/behavior.go

```diff
@@ -199,23 +216,35 @@ type handler interface {
 	Handle(context.Context, yarpc.ReqMeta, interface{}) (interface{}, yarpc.ResMeta, error)
 }
 
-func assertBaggageMatches(t crossdock.T, ctx context.Context, want transport.Headers) bool {
+func assertBaggageMatches(t crossdock.T, ctx context.Context, want map[string]string) bool {
 	assert := crossdock.Assert(t)
-	got := baggage.FromContext(ctx)
+	got := getOpenTracingBaggage(ctx)
 
-	if want.Len() == 0 {
+	if len(want) == 0 {
 		// len check to handle nil vs empty cases gracefully.
-		return assert.Equal(0, got.Len(), "baggage must be empty: %v", got)
+		return assert.Equal(0, len(got), "baggage must be empty: %v", got)
```

Comment:
nit: you can switch this to `assert.Empty` instead. This was an `Equal` check before because `Empty` relies on `len` which isn't available on our custom `Headers` type.